examples/systemcalls/systemcalls.c

```diff
@@ -17,6 +22,16 @@ bool do_system(const char *cmd)
  *   or false() if it returned a failure
 */
 
+    // SURYA: Check if cmd is NULL?
+    
+    int retValue = system(cmd);
+    
+    // -1 if child process not created, 127 if shell could not be executed in child process
+    if(retValue == -1 || retValue == 127)
```

Comment:
@suryakanteti does this check cover all the possible error cases outlined in the function description?